batch7crud-roll4/src/main/java/com/lftechnology/batch7crud/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,42 @@
+package com.lftechnology.batch7crud.filter;
+
+import static com.lftechnology.batch7crud.constant.UrlConstants.*;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+
+/**
+ * Created by pratishshr on 1/26/16.
+ */
+public class AuthenticationFilter implements Filter {
+  private String loginUri;
+
+  @Override
+  public void init(FilterConfig filterConfig) throws ServletException {
+    loginUri = filterConfig.getInitParameter("loginUri");
+  }
+
+  @Override
+  public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+          throws IOException, ServletException {
+    HttpServletRequest request = (HttpServletRequest) servletRequest;
+    HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+    HttpSession session = request.getSession();
+    String username = (String) session.getAttribute("username");
+
+    if (username == null && !(request.getContextPath() + loginUri).equals(request.getRequestURI())) {
+      response.sendRedirect(request.getContextPath() + "/auth/login");
+    } else {
+      filterChain.doFilter(request, response);
+    }
+  }
+
+  @Override
+  public void destroy() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)